compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -67,6 +67,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    fn is_slice_ty(&self, ty: Ty<'tcx>, span: Span) -> bool {
+        self.autoderef(span, ty)
+            .any(|(ty, _)| self.probe(|_| matches!(ty.kind(), ty::Slice(..) | ty::Array(..))))
```

Comment:
I'm pretty sure the probe(...) here is not actually meaningful, right? The body is not creating any obligations, it's just checking an enum discriminant basically?

Reply:
You're right. That should be fixed.